sponsor-dapp-v2/src/lib/custom-hooks.js

```diff
@@ -6,6 +6,33 @@ import { MAX_UINT_VAL } from "common/Constants";
 import { sendGaEvent } from "lib/google-analytics";
 import { ContractStateEnum } from "common/TokenizedDerivativeUtils";
 
+// This is a hack to handle reverts for view/pure functions that don't actually revert on public networks.
+// See https://forum.openzeppelin.com/t/require-in-view-pure-functions-dont-revert-on-public-networks/1211 for more
+// info.
+export function revertWrapper(result) {
+  if (!result) {
+    return null;
+  }
+
+  let revertValue = "3963877391197344453575983046348115674221700746820753546331534351508065746944";
+  if (result.toString() === revertValue) {
+    return null;
+  }
+
+  const isObject = obj => {
+    return obj === Object(obj);
+  };
+
+  if (isObject(result)) {
+    // Iterate over the properties of the object and see if any match the revert value.
+    for (let prop in result) {
+      if (result[prop].toString() === revertValue) {
+        return null;
+      }
+    }
+  }
```

Comment:
Can you explain why you don't need to return `result` from here if there wasn't a revert?

Reply:
Nope, I just forgot to add that line. Nice catch.